### ISSUE TITLE
Judy-devel installation error

### ISSUE THREAD

**Issue by DPR1604** [netdata/netdata#9356]
##### Bug report summary

When Attempting to install netdata cloud agent I'm currently getting errors installing Judy-devel when using the kickstart script.

I have narrowed the issue down to this output 

```
yum install http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm
Extra OKay Packages for Enterprise Linux - x86_64                                                                                                                                                             1.7 MB/s | 3.7 MB     00:02
Last metadata expiration check: 0:00:02 ago on Wed Jun 17 14:05:33 2020.
[MIRROR] Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm: Status code: 404 for http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm (IP: 109.228.54.129)
[MIRROR] Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm: Status code: 404 for http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm (IP: 109.228.54.129)
[MIRROR] Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm: Status code: 404 for http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm (IP: 109.228.54.129)
[MIRROR] Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm: Status code: 404 for http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm (IP: 109.228.54.129)
[FAILED] Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm: Status code: 404 for http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm (IP: 109.228.54.129)
Status code: 404 for http://mirror.centos.org/centos/8/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm (IP: 109.228.54.129)

```
It looks like judy-devel has been removed from the centos8 official mirror

##### OS / Environment

```
/etc/centos-release:CentOS Linux release 8.2.2004 (Core)
/etc/os-release:NAME="CentOS Linux"
/etc/os-release:VERSION="8 (Core)"
/etc/os-release:ID="centos"
/etc/os-release:ID_LIKE="rhel fedora"
/etc/os-release:VERSION_ID="8"
/etc/os-release:PLATFORM_ID="platform:el8"
/etc/os-release:PRETTY_NAME="CentOS Linux 8 (Core)"
/etc/os-release:ANSI_COLOR="0;31"
/etc/os-release:CPE_NAME="cpe:/o:centos:centos:8"
/etc/os-release:HOME_URL="https://www.centos.org/"
/etc/os-release:BUG_REPORT_URL="https://bugs.centos.org/"
/etc/os-release:
/etc/os-release:CENTOS_MANTISBT_PROJECT="CentOS-8"
/etc/os-release:CENTOS_MANTISBT_PROJECT_VERSION="8"
/etc/os-release:REDHAT_SUPPORT_PRODUCT="centos"
/etc/os-release:REDHAT_SUPPORT_PRODUCT_VERSION="8"
/etc/os-release:
/etc/redhat-release:CentOS Linux release 8.2.2004 (Core)
/etc/system-release:CentOS Linux release 8.2.2004 (Core)
```

##### Component Name

Kickstart.sh

##### Steps To Reproduce

1. run bash <(curl -Ss https://my-netdata.io/kickstart.sh)

**Comment by ntlug** [comment]
This is a "bug" with the extra repos (add ons) with RHEL 8.2/CentOS 8.2.  I did not have any problem with CentOS 8.1.  Judy, unfortunately, does not appear to be part of the "normal" repo, but comes from AppStream.  But right now for CentOS 8.2, they are incongruous, looks like Judy and Judy-devel might have been built from different "trees" (or Judy-devel was simply never updated correctly).  The "workaround" if you can call it that, is to install yum-utils and use yumdownloader to pull the Judy-devel rpm directly and use rpm with --nodeps to install.  I can't say if this breaks Judy or not, but it satisfies the Netdata kickstart.sh requirements.

**Comment by prologic** [comment]
Yeah thanks @ntlug ! I can't really say for sure that I have a good solution for this at all. We keep running into issues with many systems that just simply a) don't have Judy or b) it is otherwise broken in some way.

I _think_ we should probably vendor Judy at this point. The problem here of course is that vendoring violates Debian packaging polices if you want to get into their official tress (_not sure about CentOS/RHEL_).

**Comment by DPR1604** [comment]
So I was having an extra poke around to see what else I could find this morning not tested it just yet but it appears to be part of okey repo according to this https://centos.pkgs.org/8/okey-x86_64/Judy-devel-1.0.5-18.el8.x86_64.rpm.html might be a viable workaround for centos 8.2 that doesn't involve yum-utils and yumdownloader going to give it a shot on a test VM and will get back to you.

**Comment by DPR1604** [comment]
Right so the above also didn't work however I did find the RPM still being hosted on rpmfind.net and managed to get the install working so as a temporary workaround the following works to get judy-devel install on centos 8.2

``` 
yum install https://rpmfind.net/linux/centos/8-stream/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm
```
I cant any issues so far but will keep this thread updated if I experience anything.

**Comment by prologic** [comment]
Yes where we used to get this RPM from on the CentOS mirrors I believe are now 404'ing. I don't yet have a good solution/alternative for this yet.

**Comment by ntlug** [comment]
> Right so the above also didn't work however I did find the RPM still being hosted on rpmfind.net and managed to get the install working so as a temporary workaround the following works to get judy-devel install on centos 8.2
> 
> ```
> yum install https://rpmfind.net/linux/centos/8-stream/PowerTools/x86_64/os/Packages/Judy-devel-1.0.5-18.module_el8.1.0+217+4d875839.x86_64.rpm
> ```
> 
> I can't see any issues so far but will keep this thread updated if I experience anything.

What I said does "work" (quoting intentional).  Install Judy via dnf.  Use yumdownloader to pull the Judy-devel rpm down and install with --nodeps.

Then install netdata.

The real fix is having congruent packages for Judy and Judy-devel.

**Comment by ntlug** [comment]
Querying a system which was able to do the yumdownloader approch, I get this back trying to determine what repo it got Judy-devel from (note it says repository is the "okay" repository):

```
# dnf repoquery -i --available Judy-devel
Last metadata expiration check: 1:14:36 ago on Fri 26 Jun 2020 01:27:44 PM CDT.
Name         : Judy-devel
Version      : 1.0.5
Release      : 18.el8
Architecture : x86_64
Size         : 74 k
Source       : Judy-1.0.5-18.el8.src.rpm
Repository   : okay
Summary      : Development libraries and headers for Judy
URL          : http://sourceforge.net/projects/judy/
License      : LGPLv2+
Description  : This package contains the development libraries and header files
             : for developing applications that use the Judy library.
```

**Comment by prologic** [comment]
> Edit: didn't test, the install-required-packages.sh tries to force install a non-existent Judy-devel from PowerTools. My recommendation, pull both Judy and Judy-devel from "okay". Note: may force reinstall of Judy from the "okay" repo if already installed via AppStream.

I'm okay with this. Would you like to put up a PR to this effect for `./packaging/installer/install-required-packages.sh` This _may_ also help to solve #8602

**Comment by prologic** [comment]
I just had a quick look at this. Even if the Judy and Judy-devel packages work fine from the Okay repo, we also need the PowerTools repo for some build tools which also come (_it seems_) with conflicting versions of the same Judy / Judy-devel packages.

**Comment by prologic** [comment]
PowerTools contains:

```#!sh
sh-4.4# yum info Judy.x86_64
Last metadata expiration check: 0:00:23 ago on Mon Jun 29 06:25:19 2020.
Available Packages
Name         : Judy
Version      : 1.0.5
Release      : 18.module_el8.1.0+217+4d875839
```

Which is missing `Judy-devel`; whilst Okay repo(s) contain:

```#!sh
sh-4.4# yum info Judy-devel.x86_64
Last metadata expiration check: 0:00:21 ago on Mon Jun 29 06:26:59 2020.
Available Packages
Name         : Judy-devel
Version      : 1.0.5
Release      : 18.el8
Architecture : x86_64
Size         : 74 k
Source       : Judy-1.0.5-18.el8.src.rpm
```

Which requires:

```#!sh
sh-4.4# yum install Judy-devel.x86_64
Last metadata expiration check: 0:00:57 ago on Mon Jun 29 06:26:59 2020.
Error:
 Problem: package Judy-devel-1.0.5-18.el8.x86_64 requires Judy = 1.0.5-18.el8, but none of the providers can be installed
  - conflicting requests
  - package Judy-1.0.5-18.el8.x86_64 is filtered out by modular filtering
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

I'm not yet sure how to resolve this conflict :/

**Comment by ntlug** [comment]
We're sort of messed up if Judy is already installed.

However, if not, then when you pull in the okay repo, you can install both Judy and Judy-devel from it with

`dnf install Judy Judy-devel --disablerepo=AppStream`

This is still probabilistic.  But ideally, the ones it will find will come out of the okay repo.

(this is for CentOS 8 and presumably RHEL 8)

**Comment by prologic** [comment]
When I was digging into this-further, I wasn't even able to get Netdata built on CentOX 8.x even with just the Okay repo. But maybe I _might_ have to test this again.

**Comment by ntlug** [comment]
My working (flat) ansible for CentOS 8 (we have EPEL installed as part of a different playbook):

```
---
# Note: this probably is just for "8" and not other versions.
#
- hosts: g_netdata
  vars:
    okay_repo_url: "http://repo.okay.com.mx/centos/{{ ansible_distribution_major_version }}/x86_64/release/okay-release-1-3.el{{ ansible_distribution_major_version }}.noarch.rpm"
    okay_repo_gpg_key_url: ""
    okay_repofile_path: "/etc/yum.repos.d/okay.repo"
     
  tasks:
  - name: Check if a netdata is installed.
    shell:
      cmd: "netdata -V"
    failed_when:   False
    changed_when:  False
    register: netdata_version

  - name: Check if okay repo is already configured.
    stat: path={{ okay_repofile_path }}
    register: okay_repofile_result
 
  - name: Install okay repo by rpm.
    yum:
      name: "{{ okay_repo_url }}"
      state: present
    register: result
    when: not okay_repofile_result.stat.exists

  #
  # Install Judy and open-vm-tools on VMware guests
  #
  - name: Install Judy packages
    yum:
      name:
        - Judy
        - Judy-devel
      disablerepo: "AppStream"
      state: present
 
  #
  # Install netdata via curl
  #  (note: not a lot of error handling here)
  #
  - name: Install netdata
    shell:
      cmd: "bash <(curl -Ss https://my-netdata.io/kickstart.sh) --non-interactive"
      executable: "/usr/bin/bash"
    when: netdata_version.rc != 0

```

**Comment by ntlug** [comment]
(forget that comment about open-vm-tools)

**Comment by GarlandM** [comment]
Also experiencing this issue.

**Comment by Ferroin** [comment]
This will be solved by #9859.
